### ISSUE TITLE
EnzyNet

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#683]
https://arxiv.org/abs/1707.06017
https://github.com/shervinea/enzynet

Predicts enzyme function using 3D conv net. Has a Keras implementation. Would be useful to duplicate in TensorGraph.

**Comment by rbharath** [comment]
H/T @drnate

**Comment by rbharath** [comment]
Ack, meant H/T @dr-nate

**Comment by pvskand** [comment]
@rbharath, I have read the paper and there are 2 loss functions (weighted and uniform). Weighted one is basically a class balancing technique in the cases of less data or biased data and the uniform weights give equal weights to all the 6 classes of prediction. 
It was observed that the overall accuracy for the uniform weights was performing well, so I guess first we should start with the `uniform weights loss` and if required we can add the adapted weights as well since it won't actually need much changes in the code. Any comments on this?

**Comment by rbharath** [comment]
@pvskand Yes, that sounds like a good implementation strategy!

**Comment by pvskand** [comment]
@rbharath, should we also ship a pretrained model for this?

**Comment by rbharath** [comment]
@pvskand Might be useful, but I think it's a secondary design goal likely

**Comment by Manan1602** [comment]
Hey @rbharath, I see the issue is open, should I have a go at it? I majorly use PyTorch if that'd be okay?

**Comment by harishwar017** [comment]
Hey @rbharath can I take up this as my first contribution, if it is still open?

**Comment by omeygun** [comment]
Hi @rbharath , I see this issue is open. Can I take this as my first contribution?
